### PR TITLE
[testdrive] [kafka-verify] Debug option to print topic contents

### PR DESCRIPTION
**Pull request by cjubb39**
This has often been useful when debugging and having to recompile is annoying so seems like a useful tool to have around -- especially since it doesn't make the code much more complicated

### Motivation
tests

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note): n/a

  -

### CODE REVIEW

File: src/testdrive/src/format/avro.rs [hunk 1]
```diff
@@ -260,45 +261,50 @@ where
...
-        match (expected_item, actual_item) {
-            (Some(e), Some(a)) => {
-                let e_str = format!("{:#?}", e);
-                let a_str = match &regex {
-                    Some(regex) => regex
-                        .replace_all(&format!("{:#?}", a).to_string(), regex_replacement.as_str())
-                        .to_string(),
-                    _ => format!("{:#?}", a),
-                };
+    if !debug_print_only {
```
Comment (guswynn) [inline src/testdrive/src/format/avro.rs:264]:
```suggestion
    // If `debug_print_only` is true, then we skip this block, and below print ALL the messages in the topic
    if !debug_print_only {
```

File: src/testdrive/src/format/avro.rs [hunk 2]
```diff
@@ -260,45 +261,50 @@ where
...
+                    }
+                    break;
                 }
-                break;
+                (None, None) => break,
             }
-            (None, None) => break,
         }
     }
     let expected: Vec<_> = expected.map(|e| format!("{:#?}", e)).collect();
```
Comment (guswynn) [inline src/testdrive/src/format/avro.rs:310]: do we want to print expected if `debug_print_only` is true? or just `actual?

**Review by guswynn — APPROVED** [review]
some nits but :seemsreasonable:

File: src/testdrive/src/format/avro.rs (hunk 2 above)
Comment (cjubb39) [reply src/testdrive/src/format/avro.rs:310]: whoops yeah bad cherry pick! but now that I'm looking at it again, I can simplify

**Pull request merged**